src/Components/Firebase/AuthContext.js

```diff
@@ -0,0 +1,31 @@
+import React, { createContext, useContext, useState, useEffect } from "react";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "./Firebase";
+
+const AuthContext = createContext();
+
+export const useAuth = () => {
+  return useContext(AuthContext);
+};
+
+export const AuthProvider = ({ children }) => {
+  const [authUser, setAuthUser] = useState({});
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      setAuthUser(user);
+    });
+
+    return unsubscribe;
+  }, []);
```

Comment:
this is a little odd, are we using useEffect to register some kind of change listener?